internal/net/call/call.go

```diff
@@ -385,25 +381,36 @@ func (rc *reconnectingConnection) Call(ctx context.Context, h MethodKey, arg []b
 }
 
 func (rc *reconnectingConnection) callOnce(ctx context.Context, h MethodKey, arg []byte, opts CallOptions) ([]byte, error) {
-	var hdr [msgHeaderSize]byte
-	copy(hdr[0:], h[:])
+	enc := codegen.NewEncoder()
+	copy(enc.Grow(len(h)), h[:])
+
+	var micros int64
 	deadline, haveDeadline := ctx.Deadline()
 	if haveDeadline {
 		// Send the deadline in the header. We use the relative time instead
 		// of absolute in case there is significant clock skew. This does mean
 		// that we will not count transmission delay against the deadline.
-		micros := time.Until(deadline).Microseconds()
+		micros = time.Until(deadline).Microseconds()
 		if micros <= 0 {
 			// Fail immediately without attempting to send a zero or negative
 			// deadline to the server which will be misinterpreted.
 			<-ctx.Done()
 			return nil, ctx.Err()
 		}
-		binary.LittleEndian.PutUint64(hdr[16:], uint64(micros))
 	}
+	enc.Int64(micros)
 
 	// Send trace information in the header.
-	writeTraceContext(ctx, hdr[24:])
+	writeTraceContext(ctx, enc)
+
+	// Send context metadata in the header.
+	writeContextMetadata(ctx, enc)
+
+	// Note that we send the header and the payload as follows:
+	// [header_length][encoded_header][payload]
+	hdrSlice := make([]byte, 4)
+	binary.BigEndian.PutUint32(hdrSlice, uint32(len(enc.Data())))
```

Comment:
Why BigEndian? I think we are using LittleEndian everywhere else.

Reply:
Good point.